public/app/features/query/components/QueryActionComponent.ts

```diff
@@ -1,10 +1,11 @@
-import { DataQuery, TimeRange } from '@grafana/data';
+import { DataQuery, DatasourceRef, TimeRange } from '@grafana/data';
 
 interface ActionComponentProps {
   query?: DataQuery;
   queries?: Array<Partial<DataQuery>>;
   onAddQuery?: (q: DataQuery) => void;
   timeRange?: TimeRange;
+  dataSourceRef?: DatasourceRef;
```

Comment:
actually what makes more sense to send here is dataSource: DataSourceInstanceSettings, 
https://github.com/grafana/grafana/blob/ef4a76c8e6bfbef403294dbb9d5bc6cff6902108/public/app/features/query/components/QueryEditorRow.tsx#L42
 
That is passed to the QueryEditorRow, then you wont have to look it up again